Change tidb_opt_projection_push_down default value to ON (#17891)
pingcap/docs-cn: system-variables.md

### `tidb_opt_projection_push_down` <span class="version-mark">从 v6.1.0 版本开始引入</span>

> - 作用域：SESSION | GLOBAL

Reviewer: ```suggestion
- 作用域：SESSION | GLOBAL
- 是否持久化到集群：是/否
```

### `tidb_opt_projection_push_down` <span class="version-mark">从 v6.1.0 版本开始引入</span>

- 作用域：SESSION | GLOBAL
- 是否持久化到集群：是
- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：是
- 类型：布尔型
- 默认值：`ON`。在 v8.3.0 之前，默认值为 `OFF`。
- 指定是否允许优化器将 `Projection` 算子下推到 TiKV。开启后，优化器可能会将以下三种类型的 `Projection` 算子下推到 TiKV：
    - 算子顶层表达式全部为 [JSON 查询类函数](/functions-and-operators/json-functions/json-functions-search.md)或 [JSON 值属性类函数](/functions-and-operators/json-functions/json-functions-return.md)，例如 `SELECT JSON_EXTRACT(data, '$.name') FROM users;`。
    - 算子顶层表达式部分为 JSON 查询类函数或 JSON 值属性类函数，部分为直接的列读取，例如 `SELECT JSON_DEPTH(data), name FROM users;`。
    - 算子顶层表达式全部为直接的列读取，且输出的列数量小于输入的列数量，例如 `SELECT name FROM users;`。

> - `Projection` 算子最终下推与否，还取决于优化器对性能的综合评估。

Reviewer 2: ```suggestion
- `Projection` 算子最终下推与否，还取决于优化器对查询代价的综合评估。
```

> - 对于从 v8.3.0 以前的版本升级到 v8.3.0 或更新版本的 TiDB 集群，该选项默认关闭。

Reviewer 2: 对于从 v8.3.0 以前的版本升级到 v8.3.0 或更新版本的 TiDB 集群，该选项是默认关闭呢，还是保持升级前的设置呢？（比如升级前设置为了 ON，升级后是 ON 吗）

Author: 升级前不可能为 ON 的，这个 feature 之前有功能问题

> - 对于从 v8.3.0 以前的版本升级到 v8.3.0 或更新版本的 TiDB 集群，该选项默认关闭。

Reviewer 2: ```suggestion
- 对于从 v8.3.0 以前的版本升级到 v8.3.0 或更新版本的 TiDB 集群，该变量将保持升级前的配置。
```

> - 对于从 v8.3.0 以前的版本升级到 v8.3.0 或更新版本的 TiDB 集群，该变量将为 `OFF`。

Reviewer 2: ```suggestion
- 对于从 v8.3.0 以前的版本升级到 v8.3.0 或更新版本的 TiDB 集群，该变量将默认为 `OFF`。
```

> - 对于从 v8.3.0 以前的版本升级到 v8.3.0 或更新版本的 TiDB 集群，该变量将保持升级前的配置。

Author: ```suggestion
- 对于从 v8.3.0 以前的版本升级到 v8.3.0 或更新版本的 TiDB 集群，该变量将为 `OFF`。
```

- 作用域：SESSION
- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：是
- 类型：布尔型

> - 默认值：`ON`

Reviewer: ```suggestion
- 默认值：`ON`。在 v8.3.0 之前，默认值为 `OFF`。
```

> - 指定是否允许优化器将 `Projection` 算子下推到 TiKV。目前只有三种类型的 `Projection` 算子被允许下推：

Reviewer: ```suggestion
- 指定是否允许优化器将 `Projection` 算子下推到 TiKV。开启后，优化器可能会将以下三种类型的 `Projection` 算子下推到 TiKV：
```

> - 指定是否允许优化器将 `Projection` 算子下推到 TiKV。目前只有三种类型的 `Projection` 算子被允许下推：
>     - 算子顶层表达式全部为 JSON 查询类或者 JSON 值属性类函数, 如 `JSON_EXTRACT`, `JSON_DEPTH` 等。

Reviewer: ```suggestion
- 指定是否允许优化器将 `Projection` 算子下推到 TiKV。开启后，优化器可能会将以下三种类型的 `Projection` 算子下推到 TiKV：
    - 算子顶层表达式全部为 JSON 查询类或者 JSON 值属性类函数，例如 `JSON_EXTRACT()`、`JSON_DEPTH()` 等。
```

>     - 算子顶层表达式全部为 JSON 查询类或者 JSON 值属性类函数, 如 `JSON_EXTRACT`, `JSON_DEPTH` 等。
>     - 算子顶层表达式部分为 JSON 查询类或者 JSON 值属性类函数，部分为直接的列读取。
>     - 算子顶层表达式全部为直接的列读取，且输出的列数量小于输入的列数量。

Reviewer: ```suggestion
    - 算子顶层表达式全部为 [JSON 查询类函数](/functions-and-operators/json-functions/json-functions-search.md)或 [JSON 值属性类函数](/functions-and-operators/json-functions/json-functions-return.md)，例如 `SELECT JSON_EXTRACT(data, '$.name') FROM users;`。
    - 算子顶层表达式部分为 JSON 查询类函数或 JSON 值属性类函数，部分为直接的列读取，例如 `SELECT JSON_DEPTH(data), name FROM users;`。
    - 算子顶层表达式全部为直接的列读取，且输出的列数量小于输入的列数量，例如 `SELECT name FROM users;`。
```

> - `Projection` 算子最终下推与否，还取决于优化器对于性能的综合评估。

Reviewer: ```suggestion
- `Projection` 算子最终下推与否，还取决于优化器对性能的综合评估。
```
